pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -51,8 +51,11 @@ type SpdyRoundTripper struct {
 	// conn is the underlying network connection to the remote server.
 	conn net.Conn
 
-	// Dialer is the dialer used to connect.  Used if non-nil.
-	Dialer *net.Dialer
+	// dialer handle dialing for this round trip
+	dialer Dialer
```

Comment:
I didn't look at the other tests that make use of this, but storing the dial func doesn't help much because of how it's used [here](https://github.com/kubernetes/kubernetes/pull/21185/files?diff=split#diff-d6dac5243e58c2adf92dedf7de74ff1fR229). 

So I think we either extract the entire dial strategy, or keep it as is. The actual dial logic didn't change in the extracted version, `SecureDialer#dial` is pretty much what `dialWithouthProxy` was. Tests can make use of `SecureDialer` directly (which can also take custom `*net.Dialer`) or write a new, which is what I was planning to do to [mock the proxy](https://github.com/kubernetes/kubernetes/pull/21185/files?diff=split#diff-399de7ad04a2738396508d1817350e4fR232).

No strong preference and I get the non-test changes point, but I'd say we can't cover the entire `SpdyRoundTripper#dial` without making a couple things injectable.